middleware/auth.js

```diff
@@ -0,0 +1,18 @@
+import { parseCookies } from 'h3'
```

Comment:
I don't think that you need this library. Specially only for cookie parsing.
You written cookie parser you self in getUserCookie just make it more dinamic.